tests/test_registrations/test_embargoes.py

```diff
@@ -1096,3 +1097,46 @@ def test_non_contributor_GET_disapproval_returns_HTTPError(self):
         assert_equal(http.UNAUTHORIZED, res.status_code)
         assert_true(self.registration.is_pending_embargo)
         assert_equal(self.registration.embargo.state, Embargo.UNAPPROVED)
+
+class TestEmbargoUnauthView(OsfTestCase):
+
+    def setUp(self):
+        super(TestEmbargoUnauthView, self).setUp()
+
+        self.user = AuthUserFactory()
+        self.non_contrib = AuthUserFactory()
+
+        self.project = ProjectFactory(creator=self.user)
+        self.registration_embargo = RegistrationFactory(creator=self.user, project=self.project)
+
+        self.registration_embargo.embargo_registration(
+            self.user,
+            timezone.now() + timedelta(days=10)
+        )
+        self.registration_embargo.save()
+
+    def test_pending_embargo_non_contrib_returns_forbidden(self):
+        res = self.app.get(
+            self.registration_embargo.web_url_for('view_project'),
+            auth=self.non_contrib.auth,
+            expect_errors=True
+        )
+
+        assert_true(self.registration_embargo.is_pending_embargo)
+        assert_equal(res.status_code, 403)
+        assert_in('Forbidden', res.body)
+
+    def test_embargo_non_contrib_returns_resource_under_embargo(self):
+        approval_token = self.registration_embargo.embargo.approval_state[self.user._id]['approval_token']
```

Comment:
@fabmiz It looks like it is needed to me-- the first test is for pending embargo, which should throw Forbidden, the second test is for after the embargo is approved, which then throws the custom embargo error.